test/lib/classify/cache_test.rb

```diff
@@ -5,9 +5,60 @@
 class PrimerClassifyCacheTest < Minitest::Test
   def test_clear_clears_the_cache
     Primer::Classify::Cache.instance.clear!
+    Primer::Classify::AttrCache.instance.clear!
 
-    assert_empty Primer::Classify::Cache.instance.instance_variable_get(:@lookup)
+    assert Primer::Classify::Cache.instance.empty?
+    assert Primer::Classify::AttrCache.instance.empty?
   ensure
-    Primer::Classify::Cache.instance.preload!
+    Primer::Classify::AttrCache.instance.preload!
+  end
+
+  def test_evicts_entries
+    # rubocop:disable Style/RedundantFetchBlock
+    cache = Primer::Classify::Cache.instance
+    lookup = cache.instance_variable_get(:@lookup)
+
+    cache.max_size = 3
+
+    cache.fetch(:foo) { :foo }
+    cache.fetch(:bar) { :bar }
+    cache.fetch(:baz) { :baz }
+
+    assert_equal cache.size, 3
+    assert_includes lookup, [:foo].hash
+
+    cache.fetch(:boo) { :boo }
+
+    assert_equal cache.size, 3
+    assert_includes lookup, [:bar].hash
+    assert_includes lookup, [:baz].hash
+    assert_includes lookup, [:boo].hash
+    refute_includes lookup, [:foo].hash
+
+    cache.max_size = Rails.application.config.primer_view_components.max_classify_cache_size
```

Comment:
Thoughts on extracting this into a setup+teardown?

```ruby
def setup
  @original_max_size = Rails.application.config.primer_view_components.max_classify_cache_size
end

def teardown
  Rails.application.config.primer_view_components.max_classify_cache_size = @original_max_size 
end
```

Reply:
Yeah, nice!